presto-main/src/main/java/com/facebook/presto/server/testing/TestingPrestoServer.java

```diff
@@ -90,6 +98,56 @@
     private final ServiceSelectorManager serviceSelectorManager;
     private final Announcer announcer;
     private final QueryManager queryManager;
+    private final GracefulShutdownHandler gracefulShutdownHandler;
+    private final ShutdownAction testShutdownAction;
+    private final boolean coordinator;
+
+    public static class TestShutdownAction implements ShutdownAction
+    {
+        private Lock lock = new ReentrantLock();
+        private Condition shutdownCalled = lock.newCondition();
+
+        @GuardedBy("lock")
```

Comment:
You can just make this `@GuardedBy("this")` if you make the change above to use a `CountDownLatch`